pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -141,50 +145,60 @@ func clusterFromProject(project string) (string, string) {
 }
 
 // validUpdateFields checks if the fields being changed are valid update fields.
-func validateUpdateFields(oldPRTB, newPRTB *apisv3.ProjectRoleTemplateBinding) error {
-	var invalidFieldName string
+func validateUpdateFields(oldPRTB, newPRTB *apisv3.ProjectRoleTemplateBinding, fieldPath *field.Path) *field.Error {
 	switch {
 	case oldPRTB.RoleTemplateName != newPRTB.RoleTemplateName:
-		invalidFieldName = "referenced roleTemplate"
+		return field.Forbidden(fieldPath.Child("roleTemplateName"), "")
```

Comment:
Add a message for your `Forbidden` errors telling the user what they did wrong.
```suggestion
		return field.Forbidden(fieldPath.Child("roleTemplateName"), "can not update")
```

Reply:
Actually, after looking at the K8s code this is a more standard way of rejecting immutable fields
 ``` golang
 field.Invalid(fieldPath.Child("roleTemplateName"), newPRTB.RoleTemplateName, "field is immutable"))
 ```